src/foundry/client/pixi/layers/controls/ruler.d.ts

```diff
@@ -125,44 +214,72 @@ declare global {
     protected _getMovementToken(): InstanceType<ConfiguredObjectClassForName<"Token">> | null | undefined;
 
     /**
-     * A helper method to return an Array of Ray objects constructed from the waypoints of the measurement
-     * @param waypoints   - An Array of waypoint `{x, y}` Objects
-     * @param destination - An optional destination point to append to the existing waypoints
-     * @returns An Array of Ray objects which represent the segemnts of the waypoint path
+     * Test whether a Token is allowed to execute a measured movement path.
+     * @param token - The Token being tested
+     * @returns Whether the movement is allowed
+     * @throws                    A specific Error message used instead of returning false
      */
-    protected _getRaysFromWaypoints(waypoints: PIXI.Point[], destination?: PIXI.Point): Ray[];
+    protected _canMove(token: Token): true | never;
```

Comment:
I'm debating what to do about the fact that `true | never` simplifies to `true`.

I don't think the `| never` will show up to end users. If it does it serves as a decent bit of documentation but there's also `@returns`. I think you can just simplify it to `true` but if you check intellisense and it's helpful then we can leave it in.